### DISCUSSION TITLE
HUE suppress color transition from off state

### DISCUSSION THREAD

**Discussion by matrixdesert** [Koenkk/zigbee2mqtt#13156] [Devices]
Hello!

i stumbled over some postings about this topic at the HA community forum and at the old zigbee2mqtt forum.
But i still have no clue if this is possible.

My hue light is turned on - say it is color green - i turn it off - i turn it on but set the color to red - the light transitions from green to red!
But the settings for transition only allow setting the time for the transition.

Is it possible to turn the light on ignoring the previous color!?

Best regards!

**Comment by mwerline** [comment]
I'm looking for a solution to this as well. I have automations where after "bedtime" I want a light (or group of lights) to come on at a low brightness and using the red color. Currnetly if they had been on earlier in the day they initially turn on at the prior color mode/color temp, then transition to red. During the day this isn't really an issue, but at night that normally means a flash of white light, which is undoing what I am trying to achieve. 

I know it is possible to start up the hue bulbs with the desired color when turning on, as I used to do this with the Hue Bridge. Somehow Hue was able to turn the bulbs on with the desired color and not fade to it after turning on... So there has to be a way. Has anyone figured out how to do this using Z2M?

I've tried "pre-staging" the color before sending a brightness or on command, but it doesn't appear the bulb pays any attention to this when off. Any help or ideas?

**Reply by sjorge** [reply]
At least on the newer hue bulbs (With BT) You can do it by setting the color_options.execute_if_off and first sending the new color and then send the state in the following message.

```
bname=light/office/mood/bulb
# NOTE: you only need to set the color_options once, after this the bulb will change color when it's off.
mosquitto_pub -t zigbee2mqtt/${bname}/set/color_options -m '{"execute_if_off": true}'
sleep 1
mosquitto_pub -t zigbee2mqtt/${bname}/set/state -m 'ON'
mosquitto_pub -t zigbee2mqtt/${bname}/set/color -m '{"hue": 0, "saturation": 100}'
mosquitto_pub -t zigbee2mqtt/${bname}/set/state -m 'OFF'
sleep 5
mosquitto_pub -t zigbee2mqtt/${bname}/set/color -m '{"hue": 128, "saturation": 100}'
mosquitto_pub -t zigbee2mqtt/${bname}/set/state -m 'ON'
```

This example sets the color_options (only need to do this once), turns on the bulb and sets it to red, then turns it off, then turns it green and after that turns it on again (now directly to green)

**Reply by mwerline** [reply]
Thanks! I just gave that a shot and it worked perfectly!!! My eyes in the middle of the night thank you!

**Reply by gitnew12** [reply]
First, thank you sjorge for a solution to this issue.

Second, considering this thread is over 2 years old, I was wondering if this is still the recommended method or is there another way in Home Assistant?

Appreciate the help in advance!

**Reply by stephanGarland** [reply]
@gitnew12 please see [this comment by me](https://github.com/basnijholt/adaptive-lighting/issues/996#issuecomment-4526194785) for an alternative way to fix it - it's essentially the above code, but automated.
